backend/src/routes/api.ts

```diff
@@ -117,6 +120,69 @@ api.get("/v1/users/:id", async (req, res, next) => {
     next(err);
   }
 });
+api.get("/v1/users/", async (req, res, next) => {
+  try {
+    const user = req.user;
+    res.json(user);
+  } catch (err) {
+    next(err);
+  }
+});
+api.post(
```

Comment:
From an API semantic point of view, this endpoint path reads like a user-create instead. For updating a user object, it would make more sense to have a user ID parameter to determine the correct user to modify.

It is easier to understand the rest of the APIs when we implement collection lookup rather than single-user fetch on the `/api/v1/users` endpoint. This helps when you need to implement a feature that uses a data endpoint that someone else implemented—you can thus rely on the resource to behave the same. This is why the other resources (projects, techs, applications) are implemented in a similar manner.

```ts
// notation: endpoint path -> response interface

// special endpoint that returns information about login session (the current user by cookie)
GET /api/v1/current-session -> { user: IUser, isLoggedIn: boolean }
// collection lookup (fetch N number of users)
GET /api/v1/users -> IUser[]
// user creation (we do not currently have a feature that needs this)
POST /api/v1/users -> User
// fetch a single user
GET /api/v1/users/{id} -> User
// modify the data of a single user
POST /api/v1/users/{id} -> User


Applications resource as an example:
// get a list of applications
GET /api/v1/applications
// create application
POST /api/v1/applications
// update a single application by ID
POST /api/v1/applications/{id}
// fetch a single application by ID
GET /api/v1/applications/{id}
```

See https://restfulapi.net/resource-naming/ -> this article describes 


This does mean that updating the user in the settings component requires two network calls. First to get information about the current logged-in user's session, which contains the user's ID. Second to update the user by ID. But this is okay, because the first call is done when the page is first loaded (via `useSession` hook) and the user ID can then be passed to the component that needs to do the update.

Example:
```ts
function UpdateDisplayNameComponent() {
  const { loading, user } = useSession(); // First call to GET /api/v1/current-session
  const [displayName, setDisplayName] = useState("");

  const updateDisplayName = async (id: string, displayName: string) => {
    await fetch("/api/v1/users/" + id, {
      method: "POST",
      headers: {"content-type":"application/json"},
      body: JSON.stringify({
        displayName: displayName
      })
    })
  }

  return (
    <>
    <input
      type="text"
      placeholder="Display Name"
      value={displayName}
      onChange={(e) => setDisplayName(e.target.value)}
    />
    <button onClick={() => updateDisplayName(displayName)}>Update</button>
    </>
  )
}
```